packages/ui/v2/form/fields.tsx

```diff
@@ -33,6 +40,91 @@ export function Label(props: JSX.IntrinsicElements["label"]) {
   );
 }
 
+function HintError(props: {
+  hintErrors?: string[];
+  fieldErrors: { [x: string]: any };
+  formState: FormState<FieldValues>;
+  fieldName: string;
+}) {
+  const { t } = useLocale();
+  const { hintErrors, fieldErrors, formState, fieldName } = props;
+  console.log(fieldName, { hintErrors, fieldErrors, formState });
+  if (hintErrors === undefined) {
```

Comment:
Together to you question if the `return <></>` is ever reached, it is a requirement for TS to expect in any case an Element, otherwise this error shows up: 

<kbd><img width="651" alt="Screenshot 2022-08-02 at 14 50 18" src="https://user-images.githubusercontent.com/467258/182441112-98d4db0e-6400-4b22-8d90-11e6550f493f.png"></kbd>

Any advice to tackle early returns and avoid that error?

Reply:
See 6187a2abb1432e45f4e59132f75ff3a631341ad8